component/core/src/main/java/org/exoplatform/social/core/upgrade/UserPasswordHashMigration.java

```diff
@@ -0,0 +1,157 @@
+          User user = picketLinkIDMService.getIdentitySession().getPersistenceManager().findUser(userName);
+          if (user != null && picketLinkIDMService.getExtendedAttributeManager().getAttribute(userName, PASSWORD_SALT_USER_ATTRIBUTE) == null) {
+            picketLinkIDMService.getExtendedAttributeManager().addAttribute(userName, PASSWORD_SALT_USER_ATTRIBUTE, saltString);
+            picketLinkIDMService.getExtendedAttributeManager().updatePassword(user, passwordHash);
+          }
+          int current = updatedPasswords.incrementAndGet();
+          if (current % CHUNCK == 0) {
+            if (picketLinkIDMService.getIdentitySession().getTransaction() != null) {
+              picketLinkIDMService.getIdentitySession().getTransaction().commit();
+            }
+            LOG.info("{}/{} passwords have been updated ({} ms for the chunck)",
+                     current,
+                     totalSize,
+                     System.currentTimeMillis() - chunckStartTime.get());
+            chunckStartTime.set(System.currentTimeMillis());
+          }
+        } catch (Exception e) {
+          userWithErrors.add(userName);
+          LOG.error("Error while creating attribute salt and updating password hash for user : {}", userName, e);
+        }
+      });
+
+      LOG.info("End upgrade of users passwords hashing algorithm. {} passwords has been updated. {} users where not treated due to error. It took {} ms",
+               totalSize - userWithErrors.size(),
+               userWithErrors.size(),
+               (System.currentTimeMillis() - startupTime));
+      if (!userWithErrors.isEmpty()) {
+        LOG.error("Users with not modified password : {}", userWithErrors);
+        throw new IllegalStateException("UserPasswordHashMigration upgrade failed due to previous errors");
+      }
+    } catch (Exception e) {
+      LOG.error("Error while getting old users passwords hash", e);
+      throw new IllegalStateException("UserPasswordHashMigration upgrade failed due to previous errors");
+    } finally {
+      RequestLifeCycle.end();
+    }
+
+  }
+
+  @Override
+  public boolean shouldProceedToUpgrade(String newVersion,
+                                        String previousGroupVersion,
+                                        UpgradePluginExecutionContext previousUpgradePluginExecution) {
+    try {
+      return picketLinkIDMService.getExtendedAttributeManager()
+                                 .getDefaultCredentialEncoder()
+                                 .getClass()
+                                 .getName()
+                                 .equals(DEFAULT_ENCODER)
+          && super.shouldProceedToUpgrade(newVersion, previousGroupVersion, previousUpgradePluginExecution);
+    } catch (Exception e) {
+      LOG.error("Error while checking current default credential encoder", e);
+      return false;
+    }
+  }
+
+  private byte[] generateRandomSalt() {
```

Comment:
Can you please reuse this method from a centralized manner to ensure to be compliant with configured algorithm in PLIDM.